rrtmgp/mo_gas_optics_rrtmgp.F90

```diff
@@ -1382,6 +1401,64 @@ function get_col_dry(vmr_h2o, plev, latitude) result(col_dry)
   end function get_col_dry
   !--------------------------------------------------------------------------------------------------------------------
   !
+  ! Compute a transport angle that minimizes flux errors at surface and TOA based on empirical fits
+  !
+  function compute_optimal_angles(this, optical_props, optimal_angles) result(err_msg)
+    ! input
+    class(ty_gas_optics_rrtmgp),  intent(in   ) :: this
+    class(ty_optical_props_arry), intent(in   ) :: optical_props
+    real(wp), dimension(:,:),     intent(  out)  :: optimal_angles
+    character(len=128)                           :: err_msg
+    !----------------------------
+    integer  :: ncol, nlay, ngpt, nbnd
+    integer  :: icol, ilay, igpt, bnd
+    real(wp) :: t, trans_total(optical_props%get_ncol(), optical_props%get_ngpt())
+    !----------------------------
+    ncol = optical_props%get_ncol()
+    nlay = optical_props%get_nlay()
+    ngpt = optical_props%get_ngpt()
+    nbnd = optical_props%get_nband()
+
+    err_msg=""
+    if(.not. this%gpoints_are_equal(optical_props)) &
+      err_msg = "gas_optics%compute_optimal_angles: optical_props has different spectral discretization than gas_optics"
+    if(.not. extents_are(optimal_angles, ncol, ngpt)) &
+      err_msg = "gas_optics%compute_optimal_angles: optimal_angles different dimension (ncol)"
+    if (err_msg /=  "") return
+
+    !
+    ! column transmissivity
+    !
+    !$acc enter data create(trans_total)
+    !$acc parallel loop gang vector collapse(2)
+    do icol = 1, ncol
+      do igpt = 1, ngpt
+        !
+        ! Column transmissivity
+        !
+        t = 0._wp
+        do ilay = 1, nlay
+          t = t + optical_props%tau(icol,ilay,igpt)
+        end do
+        trans_total(icol,igpt) = exp(-t)
+      end do
+    end do
+
+    !$acc parallel loop gang vector collapse(2)
```

Comment:
Is there a reason to have these two loops separate? Seems like it'd be both easy and clear (and memory friendly) to combine the calculation of total column transmission with the calculation of optimal angle.